### ISSUE TITLE
NumberConversion should suggest Integer(x, 10)

### ISSUE THREAD

**Issue by shalecraig** [bbatsov/rubocop#5598]
In PR #5147, I suggested that `NumberConversion` should recommend `Integer(x, 10)` instead of `Integer(x)`. Copying into an issue so we don't forget:

--------

## Context

[The comment](https://github.com/bbatsov/rubocop/pull/5147#issuecomment-358146988) is repeated below:

> Hey there! This cop isn't correct. We should be suggesting `Integer(x, 10)` instead :)
>
> `Integer` assumes a radix. If a string input starts with an `0`, it chooses base 8. For example, `Integer('0100') == 64`.
>
> Unlike `Integer()`, `.to_i` assumes a base-10 radix, so `'0100'.to_i == 100`

## Expected behavior

> Replace unsafe number conversion with number class parsing, instead of using '1'.to_i, use stricter Integer('1', 10).

## Actual behavior

> Replace unsafe number conversion with number class parsing, instead of using '1'.to_i, use stricter Integer('1').

## Steps to reproduce the problem

In `spec/rubocop/cop/lint/number_conversion_spec.rb`, add:

```ruby
it 'is a new test' do
  expect_offense(<<-RUBY.strip_indent)
    string_value = '10'
    string_value.to_i
    ^^^^^^^^^^^^^^^^^ Replace unsafe number conversion with number class parsing, instead of using string_value.to_i, use stricter Integer(string_value, 10).
  RUBY
end

```
